src/components/PostsList.tsx

```diff
@@ -1,86 +1,74 @@
 import React from 'react';
+import { Post } from '../types/Post';
+import classNames from 'classnames';
 
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+type Props = {
+  userPosts: Post[];
+  selectedPost: Post | null;
+  setShowForm: (value: boolean) => void;
+  setSelectedPost: (postId: Post | null) => void;
+};
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <th> </th>
-        </tr>
-      </thead>
+export const PostsList: React.FC<Props> = ({
+  userPosts,
+  selectedPost,
+  setShowForm,
+  setSelectedPost,
+}) => {
+  const openPostFunction = (postToSelect: Post) => {
+    if (selectedPost && selectedPost.id === postToSelect.id) {
+      setSelectedPost(null);
+    } else {
+      setSelectedPost(postToSelect);
+      setShowForm(false);
+    }
+  };
 
-      <tbody>
-        <tr data-cy="Post">
-          <td data-cy="PostId">17</td>
+  const isOpenFunction = (currentPost: Post) => {
+    if (selectedPost && selectedPost.id === currentPost.id) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
```suggestion
    return selectedPost && selectedPost.id === currentPost.id
```

it will be the same.